src/Compilers/Core/Portable/Diagnostic/Diagnostic_SimpleDiagnostic.cs

```diff
@@ -146,40 +146,6 @@ public override IReadOnlyList<Location> AdditionalLocations
                 get { return _properties; }
             }
 
-            public override bool Equals(Diagnostic obj)
```

Comment:
This change will break all the diagnostic analyzer hosts, especially IDE diagnostic service. I think there are lots of places where IDE uses hashset/immutable hashset and expects equivalent diagnostics to be filtered out. If we are changing Diagnostic.Equals to now check for reference equality, you need to look at places where we use such hashsets and use CommonDiagnosticComparer or similar comparer to avoid any breaks.